### ISSUE TITLE
"Nuget pack" does not find ".nuget\nuget.config"

### ISSUE THREAD

**Issue by TiltonJH** [NuGet/Home#1966]
"Nuget pack" ignores dependencies from packages.config because it does not find the `.nuget\nuget.config`

(The issue which is also somewhat explained here [http://nuget.codeplex.com/workitem/3097](http://nuget.codeplex.com/workitem/3097) in the comments section still exists in nuget 3.3.0.212)

**Steps to reproduce:**
My solution layout looks like the below:

```
Dev\
    .nuget\
        nuget.config
    InternalBin\
    NuGetPackages\
    MyProj1\
        MyProj1.nuspec
        MyProj1.csproj
        packages.config
    MyProj2\
        MyProj2.nuspec
        MyProj2.csproj
        packages.config
    [...]
    MyProj.sln
```

Content of the `nuget.config`:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <config>
        <add key="repositoryPath" value=".\..\NuGetPackages" />
        [...]
    </config>
    [...]
<configuration>
```

**What I do:**
I call `pack "Dev/MyProj1/MyProj1.csproj" -BasePath "Dev/InternalBin" -OutputDirectory "Dev/InternalBin" -IncludeReferencedProjects -Verbosity detailed`. (This is an example. My paths in the real thing are fully qualified.)

Nuget states that it found the `packages.config` and will add the dependencies to the resulting `*.nupkg`. - But it doesn't.

(However, the `-IncludeReferencedProjects` work as designed and lists `MyProj2` as a dependency of `MyProj1`.)

**There are workarounds to this issue:**
1. Copy the `nuget.config` to `Dev\` and adjust the `repositoryPath` accordingly. Nuget pack now seems to find the repository and the dependencies specified within the `packages.config` are added to the `*.nupkg`.
2. Copy the `nuget.config` to `Dev\MyProj\`. Nuget pack now seems to find the repository and the dependencies specified within the `packages.config` are added to the `*.nupkg`.
3. Don't specify `<add key="repositoryPath" value=".\..\NuGetPackages" />` within the `nuget.config` and therefore the default behavior of `Dev\packages\` is used.

**In conclusion `Nuget pack` does not seem to find the `Dev\.nuget\nuget.config`.**

**Comment by yishaigalatzer** [comment]
Yes, that is by design, the rule is look up from the folder you run in, and not look down. the .nuget\nuget.config does not conform to that rule.

**Comment by TiltonJH** [comment]
Well, either you are right and it is by design, which means the documentation is wrong. Or it is the other way around.
[Documentation: NuGet config extensibility point](https://docs.nuget.org/Consume/NuGet-Config-File#nuget-config-extensibility-point).

**Comment by yishaigalatzer** [comment]
@TiltonJH thanks for pointing this out, I think there is definitely a confusion here. The .nuget folder was intended for supporting solution level packages only.

While we need to figure this out, I don't see us getting to it right away, I'm also afraid of potential side effects of this on existing users that might have a nuget.config in that folder where it was never active and all of the sudden it is going to start changing behaviors for existing projects. I'm parking this issue in vNext and I suggest that you move the nuget.config to the dev folder for now.

**Comment by lukefan6** [comment]
I found it confusing, too. 

Since .nuget folder and its content are generated by visual studio template, I think there are lots of people, including @TiltonJH and me, would assume my solution nuget config went there and should be recognized by nuget command.

Hope it could be fixed, cheers.

**Comment by emgarten** [comment]
Answered above, closing.
